src/AppInstallerCommonCore/AppInstallerTelemetry.cpp

```diff
@@ -132,6 +175,33 @@ namespace AppInstaller::Logging
         m_caller = caller;
     }
 
+    void TelemetryTraceLogger::SetExecutionStage(uint32_t stage) noexcept
+    {
+        m_executionStage = stage;
+    }
+
+    std::unique_ptr<TelemetryTraceLogger> TelemetryTraceLogger::CreateSubTraceLogger() const
+    {
+        THROW_HR_IF(HRESULT_FROM_WIN32(ERROR_INVALID_STATE), !this->m_isInitialized);
+
+        auto subTraceLogger = std::make_unique<TelemetryTraceLogger>();
+
+        subTraceLogger->m_parentActivityId = this->m_activityId;
+        subTraceLogger->m_subExecutionId = s_subExecutionId++;
+
+        // Copy remaining fields from parent.

```

Comment:
We should start with the copy constructor, then modify from there.  That way we don't miss new fields.  If necessary, override the copy constructor for the summary event to simply default construct instead.